README.md

```diff
@@ -4,4 +4,67 @@ AWS Crypto Abstraction Layer: Cross-Platform, C99 wrapper for cryptography primi
 
 ## License
 
-This library is licensed under the Apache 2.0 License. 
+This library is licensed under the Apache 2.0 License.
+
+## Supported Platforms
+* Windows (Vista and Later)
+* Apple
+* Unix (via. OpenSSL compatible libcrypto)
+
+## Currently provided algorithms
+
+### Hashes
+#### MD5
+##### Streaming
+````
+struct aws_hash *hash = aws_md5_new(allocator);
+aws_hash_update(hash, &your_buffer);
+aws_hash_finalize(hash, &output_buffer, 0);
+aws_hash_destroy(hash);
+````
+
+##### One-Shot
+````
+aws_md5_compute(allocator, &your_buffer, &output_buffer, 0);
+````
+
+#### SHA256
+##### Streaming
+````
+struct aws_hash *hash = aws_sha256_new(allocator);
+aws_hash_update(hash, &your_buffer);
+aws_hash_finalize(hash, &output_buffer, 0);
+aws_hash_destroy(hash);
+````
+
+##### One-Shot
+````
+aws_sha256_compute(allocator, &your_buffer, &output_buffer, 0);
+````
+
+### HMAC
+#### SHA256 HMAC
+##### Streaming
+````
+struct aws_hmac *hmac = aws_sha256_hmac_new(allocator, &secret_buf);
+aws_hmac_update(hmac, &your_buffer);
+aws_hmac_finalize(hmac, &output_buffer, 0);
+aws_hmac_destroy(hmac);
+````
+
+##### One-Shot
+````
+aws_sha256_hmac_compute(allocator, &secret_buf, &your_buffer, &output_buffer, 0);
+````
+
+## FAQ
+### I want more algorithms, what do I do?
+Great! So do we! At a minimum, file an issue letting us know. If you want to file a Pull Request, we'd be happy to review and merge it when it's ready.
+### When will this be in the language specific crt packages?
```

Comment:
I would just leave this one out, even though it doesn't give dates.

Reply:
okay